src/libcrun/linux.c

```diff
@@ -1684,17 +1687,17 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
         }
       else
         {
-          char *dirname;
-          cleanup_free char *buffer = NULL;
           cleanup_close int dirfd = -1;
-          char *basename, *tmp;
+          cleanup_free char *dirname = NULL;
+          char *basename, *found;
 
-          buffer = xstrdup (device->path);
-          dirname = buffer;
+          dirname = strdup (normalized_path);
 
-          tmp = strrchr (buffer, '/');
-          *tmp = '\0';
-          basename = tmp + 1;
+          found = strrchr (dirname, '/');
+          if (found)
+            *found = '\0';
+
+          basename = found + 1;
```

Comment:
There is something interesting above this implementation, generally.

The added NULL check above is fine. However, if we ever get a NULL in `found`, then we would have a value of `1` in basename, and then the attempt to read from it will be either SIGSEGV, hopefully, or some undefined behaviour (worst case; should not happen on Linux, hopefully).

The old (original) code also has this problem.